examples/question_answering/question_answering/evaluator.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+import evaluate
+from question_answering.utils import compute_metric_bar_plot
+from question_answering.utils import compute_metric_vs_metric_plot
+from question_answering.utils import compute_score_distribution_plot
+from question_answering.utils import mean_metric
+from question_answering.workflow import GroundTruth
+from question_answering.workflow import Inference
+from question_answering.workflow import TestCase
+from question_answering.workflow import TestCaseMetrics
+from question_answering.workflow import TestSample
+from question_answering.workflow import TestSampleMetrics
+from question_answering.workflow import TestSuite
+from question_answering.workflow import TestSuiteMetrics
+from question_answering.workflow import ThresholdConfiguration
+
+from kolena.workflow import Evaluator
+from kolena.workflow import Plot
+
+bertscore = evaluate.load("bertscore")
+rouge = evaluate.load("rouge")
+
+
+class QuestionAnswerEvaluator(Evaluator):
+    """
+    The `QuestionAnswerEvaluator` transforms inferences into metrics for the question answer workflow.
+
+    For additional functionality, see the associated [base class documentation][kolena.workflow.evaluator.Evaluator].
+    """
+
+    test_samples_by_test_case: Dict[str, Dict[str, List[int]]] = {}  # test_case -> data_id -> turns
```

Comment:
Do you think we can avoid this cacheing if we use evaluate function instead?

Reply:
It could be avoided, but I will leave it as is

Reply:
Aside from avoiding caching, I think we can benefit for using evaluate function on not repeating computing test sample metrics. It's ok if you can't get to this today. I can clean it up and wrap it up.